database/scrapers/departments.ts

```diff
@@ -0,0 +1,188 @@
+import { DateTime } from "luxon";
+import { Readable } from "node:stream";
+
+import { BaseScraperModule, TaskHandle } from "#commands/db_scrape";
+import DepartmentModel from "#models/department";
+import DepartmentLinkModel from "#models/department_link";
+import FieldOfStudyModel from "#models/field_of_study";
+import FilesService from "#services/files_service";
+
+interface SourceResponse<T> {
+  data: T[];
+}
+
+interface DepartmentsDraft {
+  id: number;
+  name: string;
+  logo: string | null;
+  description: string | null;
+  code: string;
+  gradient_start: string;
+  gradient_end: string;
+  address: string;
+  betterCode: string;
+  createdAt: string;
+  updatedAt: string | null;
+}
+
+interface FieldOfStudyDraft {
+  id: number;
+  department_id: number;
+  name: string;
+  url: string | null;
+  isEnglish: boolean;
+  is2ndDegree: boolean;
+  isLongCycleStudies: boolean;
+  hasWeekendModeOption: boolean;
+  createdAt: string;
+  updatedAt: string | null;
+}
+
+interface DepartmentLinkDraft {
+  id: number;
+  department_id: number;
+  linkType: string;
+  link: string;
+  createdAt: string;
+  updatedAt: string | null;
+}
```

Comment:
these structs still have issues:
- I don't see the `createdAt`/`updatedAt` fields on any of the directus responses
- `department_id` on the department link struct is sometimes null